### ISSUE TITLE
MSBUILD : error MSB4223: A node of the required type OutOfProc could not be created.

### ISSUE THREAD

**Issue by natemcmaster** [Microsoft/msbuild#3214]
Our CI builds started failing with a bunch of errors that say.

```
MSBUILD : error MSB4223: A node of the required type OutOfProc could not be created.
```

### Steps to reproduce
```
git clone https://github.com/aspnet/Universe
git checkout 983eb37cd61bee4f97f9bc44914575a9cad150b2
build.cmd /p:SkipTests=true /p:CI=true
```

### Expected  behavior
Builds should pass. Only [difference in this commit](https://github.com/aspnet/Universe/compare/983eb37cd61bee4f97f9bc44914575a9cad150b2~1...983eb37cd61bee4f97f9bc44914575a9cad150b2) is that we updated to MSBuild version 15.7.169.51914 (via .NET Core CLI 2.1.300-preview3-008646). The previous commit builds with MSBuild  version 15.7.145.53551  (via .NET Core CLI 2.1.300-preview2-008530)

### Actual behavior
About 15-20 minutes into the build, MSBuild begins to fail with 
```
MSBUILD : error MSB4223: A node of the required type OutOfProc could not be created.
```

Internal build logs available here: http://aspnetci/viewLog.html?buildId=452289&tab=buildResultsDiv&buildTypeId=Releases_21Public_UpdateUniverse and again http://aspnetci/viewLog.html?buildId=452160&tab=buildResultsDiv&buildTypeId=Releases_21Public_UniverseCoherence.

We reverted to the 2.1.300-preview2 CLI to workaround for now.

### Environment data
`msbuild /version` output:
```
MSBuild version 15.7.169.51914 for .NET Core
```

OS info: Windows Server 2012 R2

If applicable, version of the tool that invokes MSBuild (Visual Studio, dotnet CLI, etc): dotnet CLI

cc @rainersigwald possibly related to node reuse changes?

**Comment by rainersigwald** [comment]
All on Windows, right? https://github.com/Microsoft/msbuild/commit/979ef17199fb4a0ee7d9c4a92305a6ea2bac8914 added a timeout on Windows as well as *nix but I expected it to have no effect on Windows since we would have failed to connect before the timeout had time to happen. Maybe that was wrong.

@natemcmaster Is it possible to collect additional diagnostic information that should help diagnose this? We log connection attempts and errors to files in `MSBUILDDEBUGPATH` if `MSBUILDDEBUGCOMM=1`, which should include the detailed exception we caught to produce this error.

**Comment by natemcmaster** [comment]
Yes, all on Windows. I have kicked off http://aspnetci/viewLog.html?buildId=452343 with diagnostic logging enabled. I'll ping this thread with the logs when they are available.

**Comment by rainersigwald** [comment]
Great, I'm looking at the logs now.

**Comment by rainersigwald** [comment]
Ok, I definitely see the new timeout firing in these logs. I'm going to prep privates that don't use that codepath on Windows.

**Comment by rainersigwald** [comment]
Fixed by #3215.
